account_loan_vendor_payment/wizard/account_payment_register.py

```diff
@@ -0,0 +1,40 @@
+from odoo import _, fields, models
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = "account.payment.register"
+
+    def _get_default_loan_journal_currency_id(self):
+        return self.env.company.currency_id.id
+
+    loan_payment = fields.Boolean("Open Loan")
+    check_bills_currency = fields.Boolean(default=False)
+    loan_journal_id = fields.Many2one(
+        "account.journal",
+        domain="[('company_id', '=', company_id), ('type', '=', 'general')]",
+    )
+    exchange_rate = fields.Float(default=1)
+    loan_journal_currency_id = fields.Many2one(
+        related="loan_journal_id.currency_id", string="Loan journal currency"
+    )
+
+    def action_create_payments(self):
+        if not self.loan_payment:
+            return super().action_create_payments()
+        payments = self._create_payments()
+        amount = self.amount * self.exchange_rate if self.exchange_rate else self.amount
```

Comment:
**question**: I'm probably missing a bit of context, could you explain why we're multiplying the amount by itself?

Reply:
Hi @PicchiSeba, exchange rate is an input value, so I multiply it to open a new loan using a different currency using the inserted exchange rate.

Reply:
I'm sorry i misinterpeted the expression.
At first I thought you wrote "multiply the amount by the exchange rate (if we have it), otherwise multiply it by itself"

But now a re-read the line and it's fine to me.